2025/build-tooling/demo/final/eslint.config.js

```diff
@@ -0,0 +1,54 @@
+import eslint from "@eslint/js";
+import tsEslint from "typescript-eslint";
+import globals from "globals";
+import react from "eslint-plugin-react";
+import reactHooks from "eslint-plugin-react-hooks";
+import reactRefresh from "eslint-plugin-react-refresh";
+import { fileURLToPath } from "node:url";
+import { dirname } from "node:path";
+
+const cwd = dirname(fileURLToPath(import.meta.url));
```

Comment:
we could use import.meta.dirname but not aafe to assume that everyone is on node.js 22 lready

Reply:
oh, cool! it used to be required, but is optional now

updated to use new projectService: https://typescript-eslint.io/packages/parser/#projectservice

we are not using it in AWC as it has an unfixed bug that affects us: https://devtopia.esri.com/WebGIS/arcgis-web-components/issues/543#issuecomment-5210975